From f373082ff22d0f1c9c27b6d33393b9516500d00c Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 5 Sep 2017 17:35:22 +0800
Subject: [PATCH 2511/5242] MLK-16372-4 arm64: dts: imx8qm: add cpu-freq
 cooling map

commit  8ec16539b984b9082ac1e9c4eb76795554c1d88b from
https://source.codeaurora.org/external/imx/linux-imx.git

Add i.MX8QM cpu-freq cooling support, when temperature
exceeds passive point, cpu-freq will drop to lowest
set-point. Both A53 and A72 cluster are supported.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index ce4c827..41d6f99 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -22,6 +22,7 @@
 #include <dt-bindings/soc/imx8_pd.h>
 #include <dt-bindings/pinctrl/pads-imx8qm.h>
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/thermal/thermal.h>
 
 / {
 	compatible = "fsl,imx8qm";
@@ -980,6 +981,13 @@
 					type = "critical";
 				};
 			};
+			cooling-maps {
+				map0 {
+					trip = <&cpu_alert0>;
+					cooling-device =
+					<&A53_0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
 		};
 
 		cpu-thermal1 {
@@ -998,6 +1006,13 @@
 					type = "critical";
 				};
 			};
+			cooling-maps {
+				map0 {
+					trip = <&cpu_alert1>;
+					cooling-device =
+					<&A72_0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
 		};
 
 		gpu-thermal0 {
@@ -2694,6 +2709,7 @@
 	>;
 	clocks = <&clk IMX8QM_A53_DIV>;
 	clock-latency = <61036>;
+	#cooling-cells = <2>;
 };
 
 &A72_0 {
@@ -2707,4 +2723,5 @@
 	>;
 	clocks = <&clk IMX8QM_A72_DIV>;
 	clock-latency = <61036>;
+	#cooling-cells = <2>;
 };
-- 
1.7.9.5

